<div class='alert alert-warning'>

SciPy's interactive examples with Jupyterlite are experimental and may not always work as expected. Execution of cells containing imports may result in large downloads (up to 60MB of content for the first import from SciPy). Load times when importing from SciPy may take roughly 10-20 seconds. If you notice any problems, feel free to open an [issue](https://github.com/scipy/scipy/issues/new/choose).

</div>

In [ ]:
import numpy as np
from scipy import stats
rng = np.random.default_rng()
res = stats.anderson_ksamp([rng.normal(size=50),
rng.normal(loc=0.5, size=30)])
res.statistic, res.pvalue

(1.974403288713695, 0.04991293614572478)

In [ ]:
res.critical_values

array([0.325, 1.226, 1.961, 2.718, 3.752, 4.592, 6.546])

The null hypothesis that the two random samples come from the same
distribution can be rejected at the 5% level because the returned
test value is greater than the critical value for 5% (1.961) but
not at the 2.5% level. The interpolation gives an approximate
p-value of 4.99%.


In [ ]:
samples = [rng.normal(size=50), rng.normal(size=30),
           rng.normal(size=20)]
res = stats.anderson_ksamp(samples)
res.statistic, res.pvalue

(-0.29103725200789504, 0.25)

In [ ]:
res.critical_values

array([ 0.44925884,  1.3052767 ,  1.9434184 ,  2.57696569,  3.41634856,
  4.07210043, 5.56419101])

The null hypothesis cannot be rejected for three samples from an
identical distribution. The reported p-value (25%) has been capped and
may not be very accurate (since it corresponds to the value 0.449
whereas the statistic is -0.291).

In such cases where the p-value is capped or when sample sizes are
small, a permutation test may be more accurate.


In [ ]:
method = stats.PermutationMethod(n_resamples=9999, random_state=rng)
res = stats.anderson_ksamp(samples, method=method)
res.pvalue

0.5254